### PR TITLE
Fix #22133: Ride time is incorrect for extremely low speeds

### LINKED ISSUE #22133: Ride time is incorrect for extremely low speeds
### Operating System

MacOS

### OpenRCT2 build

OpenRCT2, v0.4.11 (7c9b27f on develop)

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

_No response_

### Describe the issue

Using the Ride Vehicle Editor plugin, it is possible to set powered vehicles' max speed anywhere from 1 to 255 units. When setting vehicles to max speed to 1 or 2 units (1mph or less than 3kmh) the vehicles crawl so slowly that when returning to the station, the game registers the ride time as 1 second regardless of actual length. I tried recreating this with a normal vehicle in powered launch mode set to 0mph but the vehicle will still move faster than this threshold and register accurate ride time. It seems that in these extremely low speeds (units 1 and 2), the game just doesn't count the ride time but will if its at least 3 units of speed.

### Steps to reproduce

Set powered vehicle max speed to 1 or 2 units in the Ride Vehicle Editor plugin and test ride.
Ride time will read 1 second.

### Attachments

speed issue

### PR DESCRIPTION
**Pull request by lewyche**
Testing Tips: (Ride Vehicle Editor Plugin is required)

1. Build ride with powered vehicle (e.g., Monorail)
2. Test ride once (So it shows up in the plugin)
3. Close the ride
4. Set max speed to 1 in the plugin
5. Invalidate test results by setting and resetting the number of circuits
6. Test ride and set speed to turbo or hyper
7. Check test results

Potential improvements:

- Count the time when a train is completely stationary towards overall ride time, such as when using block breaks. (This would affect existing rides, and differ from the original game)
- Use fixed point integers to represent speeds (Suggested by @ZehMatt). 

Address #22133.

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -847,10 +847,12 @@ void Vehicle::UpdateMeasurements()
             curRide->max_speed = absVelocity;
         }
 
-        if (curRide->average_speed_test_timeout == 0 && absVelocity > 0x8000)
+        if (curRide->average_speed_test_timeout == 0)
         {
-            curRide->average_speed = AddClamp<int32_t>(curRide->average_speed, absVelocity);
-            stationForTestSegment.SegmentTime++;
+            if (absVelocity > 0x8000)
```
Comment (mrmbernardi) [inline src/openrct2/ride/Vehicle.cpp:852]: Why leave this condition in?

Comment (lewyche) [reply src/openrct2/ride/Vehicle.cpp]: You're right, we probably want them both to be under the same condition.

**Comment by mrmbernardi** [comment]
I would like someone who's more familiar with the gameplay implications of this to weigh in on the significance of this change.

**Comment by astraylife1** [comment]
Please keep open

**Comment by tupaschoal** [comment]
> I would like someone who's more familiar with the gameplay implications of this to weigh in on the significance of this change.

If it helps someone, it seems like `0x8000` is equivalent to `0.5_mph` on our code base. There are only two places considering this speed:

https://github.com/OpenRCT2/OpenRCT2/blob/e8c56e953813a644749217f9b23f7269847bc56c/src/openrct2/ride/Vehicle.cpp#L8432-L8435

https://github.com/OpenRCT2/OpenRCT2/blob/e8c56e953813a644749217f9b23f7269847bc56c/src/openrct2/ride/Vehicle.cpp#L8685-L8689

**Review by duncanspumpkin — APPROVED** [review]
Should also increment the network version

**Review by mrmbernardi — APPROVED** [review]

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**